### DISCUSSION TITLE
Is "go to definition" possible?

### DISCUSSION THREAD

**Discussion by ftc** [James-Yu/LaTeX-Workshop#3294] [Q&A]
With other languages in vs code, right clicking on a symbol has an option to "go to definition".  Is this possible with LaTeX?  e.g.

```
\newcommand{\foo}{Hi there!}

...

\foo % pressing go to definition here moves the cursor to the newcommand
```

**Comment by jlelong** [comment]
The `go to definition` feature relies on the symbols defined by the document. Currently we limit symbols to anything related to the table of contents (sections, figures, labels) and we do not plan to change. So, it is not possible to use the `go to definition` feature on commands.

**Reply by ftc** [reply]
Ah, that is unfortunate.  But I can see how resolving other symbols could be difficult given the nature of latex.

**Comment by jlelong** [comment]
My previous answer was wrong. Actually _go to definition_ should have been working but there was a bug.
This was fixed in #3296. This will be available in the next release (`> 8.26.0`).

**Reply by ftc** [reply]
Awesome, thank you @jlelong!  Looking forward to it!
